src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticSearchSink.cs

```diff
@@ -72,7 +72,7 @@ protected override void EmitBatch(IEnumerable<LogEvent> events)
                 var items = result.Body["items"];
                 foreach (var item in items)
                 {
-                    if (item["index"] != null && HasProperty(item["index"], "error") && item["index"]["error"] != null)
```

Comment:
Agreed. It should suffice to change the first condition of `HasProperty` from ` if (settings is System.Dynamic.ExpandoObject)` to `if (settings is System.Collections.IDictionary || settings is System.Dynamic.ExpandoObject)`.

I believe `item` isn't always an `IDictionary`, in fact I think I tried `.ContainsKey` first and it didn't work for me because it was some other kind of dynamic object.

No idea about how to integration test this, unfortunately. Unit testing the `HasProperty` function would be trivial of course but wouldn't help at all with ES version changes.